processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/PartitionContext.java

```diff
@@ -134,6 +136,7 @@ public void addRequest(TaskRequest request) {
     }
 
     public OffsetState registerOffset(long offset) {
+        latestConsumedOffset = Math.max(latestConsumedOffset, offset);
```

Comment:
I think there's no chance of the offset being registered to go backward? Even in regression case, we should update `latestConsumedOffset` as is because we're actually consuming that point now?